cloud/credentials.go

```diff
@@ -87,6 +87,17 @@ func NewCredential(authType AuthType, attributes map[string]string) Credential {
 	return Credential{authType: authType, attributes: copyStringMap(attributes)}
 }
 
+// CredentialFromProperties returns an immutable, Credential with the supplied
+// name, auth-type, attributes and revoke flag.
+func CredentialFromProperties(name string, authType AuthType, attributes map[string]string, revoked bool) Credential {
```

Comment:
I wonder if we should be consistent with naming here. We already have various NewCredentialXXX() methods. This is similar to the above NewCredential() except that we also have name etc. Maybe NewNamedCredential()

Reply:
done